confluent/confluent.go

```diff
@@ -57,13 +106,35 @@ func createBasicConfigMap(properties map[string]string) kafka.ConfigMap {
 	return configMap
 }
 
-// CreateTopic creates a topic using the Admin Client API.
-func CreateTopic(producer *kafka.Producer, topic string) {
+// CreateTopic creates a topic using the Admin Client API with default settings.
+// This function returns an error if the topic does not exist and cannot be
+// created for any reason.
+func CreateTopic(producer k.Producer, topic string) error {
+	topics := []string{topic}
+
+	topicCreationConfig := make(map[string]string)
+	topicCreationConfig[AdminOperationTimeout] = DefaultAdminOperationTimeout
+	topicCreationConfig[NumPartitions] = DefaultNumPartitions
+	topicCreationConfig[ReplicationFactor] = DefaultReplicationFactor
```

Comment:
Do we want to make these configurable, or will they always be set to the default?

Reply:
The 1 topic `CreateTopic` method is a convenience method that delegates to the fuller `CreateTopics` method, using defaults. If you want to specify your configuration, you can use the `CreateTopics` method directly.

Reply:
i suggest adding something for retention as that is  very important and must be determined upfront. I am not sure if  kafka has a default.

Reply:
that makes sense. thanks for the clarification